### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by acolombier**
Initial repo setup as per showcased in the [demo repo](https://github.com/acolombier/mixxx-proposals-example)

### CODE REVIEW

File: README.md
Comment (acolombier) [inline README.md]: I took the liberty to credit Prometheus community for the initial effort. Let me know if you think it is inappropriate or insufficient

File: README.md
Comment (acolombier) [inline README.md]: @mixxx/developer could you please create the `proposal` label please?

File: README.md
Comment (acolombier) [inline README.md]: I took the liberty to credit Prometheus community for the initial effort. Let me know if you think it is inappropriate or insufficient

File: .gitignore
Comment (daschuer) [inline .gitignore]: Can we clean that up for only mandatory ignores in our regular case?

File: LICENSE
Comment (daschuer) [inline LICENSE:1]: This is part of a pending discussion. An alternative would be cc-by-sa.

File: .github/workflows/proposals.yaml
Comment (daschuer) [inline .github/workflows/proposals.yaml]: What is the advantage of mdox? 

I think a pre-commit solution like in mixxxdj/mixxx works better, because the it will be formatted correctly on the side using the known workflow. 
There is also a link checker hook available.

File: 0000-00-00_template.md
Comment (daschuer) [inline 0000-00-00_template.md]: The twitter link dos not add much value. Maybe we shall copy the content literally.

File: CODE_OF_CONDUCT.md
Comment (daschuer) [inline CODE_OF_CONDUCT.md:1]: I think we should just reference https://github.com/mixxxdj/mixxx/blob/main/CODE_OF_CONDUCT.md here. 
This will release us form maintenance in case of changes.

File: Makefile
Comment (daschuer) [inline Makefile]: Can we remove this in favour of pre-commit? Maybe we can point out to use a md wysiwyg editor.

File: README.md
Comment (daschuer) [reply README.md]: Sure, but aren't all PRs proposals?

File: README.md
Comment (daschuer) [inline README.md]: Some thought, for adjusting the workflow a bit. 

The workflow proposed here, does not allow to "publish" a draft and collaborate before it is ready. You will have a proposal owner, which may accept suggestions until the proposal is "accepted."  This makes it hard when you for instance have a good idea, but no interest or time to actually implement it. 

After merge, an Owner without commit rights is no longer "Owner" other may hijack the proposal. This will not happen if the proposal is developed in the Owners repository and is only reflected as a PR in this repository. 
 
Currently we don't have a workflow for "accepting" a proposal. It is implicit accepted once you find a contributor and a reviewer for it. Both may collaborate on a proposal. How can we model this workflow?  

I think a proposal should be also a place to sort thoughts. 

In the past we had design discussions along PRs which in turn required extra work. My hope is that a proposal avoids it.
This requires that objections a phrased early. How can we archive this? 

Any ideas how we can reflect these topics as well?

File: README.md
Comment (daschuer) [inline README.md:29]: A design document shall describe the use-cases the proposal will implement and also the use-cases that are explicitly not included. 

It may also contain requirements and acceptance criteria of a derived Pull Request. This avoid feature creep and adjust the expectations during a review.

File: README.md
Comment (daschuer) [reply README.md]: That' fair.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: README.md
Comment (acolombier) [reply README.md]: I guess not. Some PRs maybe related to chores on the repo (like this one) or updates on previously approved proposal (e.g task list or status update, owner, etc...)

File: .gitignore
Comment (acolombier) [reply .gitignore]: Fair, not sure there is anything to be expected at the moment. Should I remove the git ignore entirely gor now?

File: .github/workflows/proposals.yaml
Comment (acolombier) [reply .github/workflows/proposals.yaml]: Agreed, I would also prefer pre-commit hook, but simply took what was currently setup in the template repo. Will look at replacing it.

File: README.md
Comment (daschuer) [reply README.md]: Done, and deleted all others.

File: .gitignore
Comment (daschuer) [reply .gitignore]: Yes

File: .github/workflows/proposals.yaml
Comment (daschuer) [reply .github/workflows/proposals.yaml]: Thank you.

File: LICENSE
Comment (ywwg) [reply LICENSE:1]: We are forking another project's proposal repo so we cannot change their license.

Comment (daschuer) [reply LICENSE:1]: Ok, than let's keep this license.

File: README.md
Comment (ywwg) [reply README.md]: the PR process is how proposals are discussed and accepted.  When they are merged, the proposals are accepted

**Review by ywwg — APPROVED** [review]

File: README.md
Comment (acolombier) [reply README.md]: I think the idea of the Owner here is someone who cares about driving the effort to make the proposal happen. It's isn't necessary the initial contributor (e.g cases where the original idea gets abandonned and re-openned by someone else, or case where one ask for a feature but doesn't have the time, knowledge or skills to make the idea happened).

This owner will help with:
- informing on changes on the initial plans, in case things gets adjusted after accepting a proposal
- facilitating contacts (e.g on Zulip)
- crediting the original effort on driving the change.

The advantage of writing down the owner(s) in the document is to prevent cases where the commit authors isn't the original author (as you suggested with a "hijack": this would likely lead to second owner being added, working in collaboration with the original owner, or replacing in case the former owners have lost interest in driving the feature.

Finally, "accepting" doesn't mean having the feature ready and implmented in Mixxx, but rather agreeing on making the proposal happen as time goes. For example, if the proposal repo was here at that time, the QML proposal would have already being accepted (merged) although the development effort is still ongoing. This is a way to say "Mixxx is committed to modernising the UI using QML". Does that make sense? Do you think further info  should go in the README.md?

Comment (ywwg) [reply README.md]: @daschuer do you understand the process now or is the wording still confusing?

File: README.md
Comment (ywwg) [reply README.md:29]: @acolombier I think this is a good addition

File: .github/workflows/proposals.yaml
Comment (acolombier) [reply .github/workflows/proposals.yaml]: I have used [markdownlint-cli2](https://github.com/DavidAnson/markdownlint-cli2) which I often use as it is quite flexible. Let me know if you had another in mind.

**Review by ywwg — CHANGES_REQUESTED** [review]
@acolombier let's get this in soon.  If Daniel doesn't have time to review again I think we can merge it in and make tweaks later, but let's get the ball rolling. There is one good note to address and then we can merge.

**Review by ywwg — CHANGES_REQUESTED** [review]
@acolombier let's get this in soon.  If Daniel doesn't have time to review again I think we can merge it in and make tweaks later, but let's get the ball rolling. There is one good note to address and then we can merge.

**Comment by acolombier** [comment]
> @acolombier let's get this in soon.

Sounds good to me! Apologies I haven't addressed this sooner. I just sorted the major bits (pre-commit and CI) so hopefully we are in good state now to get this in.

**Review by ywwg — APPROVED** [review]

File: 0000-00-00_template.md
Comment (Swiftb0y) [inline 0000-00-00_template.md:1]: should this template go into `./proposals` or stay here at the root?

File: README.md
Comment (Swiftb0y) [inline README.md]: it's not clear what the resulting file should be named. I assume `0000-00-00` is `YYYY-MM-DD`? Does it make sense to track? What date would that be, date of merge/acceptance or initial creation?

**Review by Swiftb0y — COMMENTED** [review]
couple questions, LGTM apart from that.

**Comment by ywwg** [comment]
@acolombier do you have time to address the last notes? let's get this in!  we can also iterate once it's set up.

File: README.md
Comment (ywwg) [inline README.md]:
```suggestion
   Make sure to use the [template](YYYY-MM-DD_template.md) as the guide for what sections should be present in the
```

File: 0000-00-00_template.md
Comment (ywwg) [reply 0000-00-00_template.md:1]: I agree, the template should go in /proposals

Comment (ywwg) [reply 0000-00-00_template.md:1]: and be renamed to YYYY-MM-DD_template.md

File: README.md
Comment (acolombier) [reply README.md]: From the README, it currently says:
> Put the creation date (the day you started preparing this design document)

I think that makes sense as it is a way to track how long the effort has been going on.

**Comment by acolombier** [comment]
@ywwg Done!

File: README.md
Comment (Swiftb0y) [reply README.md]: Ah, must've missed that. LGTM

**Review by Swiftb0y — APPROVED** [review]
I don't have a strong opinion on where the template lives. Its fine if it stays at the root so its easier to spot for people

**Comment by Swiftb0y** [comment]
Thank you for putting in the effort to set this up @acolombier

**Pull request merged**
